testing/src/scenario/context.py

```diff
@@ -745,7 +912,9 @@ def __call__(self, event: _Event, state: State) -> Manager[CharmType]:
             event: the event that the charm will respond to.
             state: the :class:`State` instance to use when handling the event.
         """
-        return Manager(self, event, state)
+        mgr = Manager(self, event, state)
+        self._manager = mgr
```

Comment:
Oh, it looks like maybe you're doing that (always creating a Manager) here?

Reply:
> Oh, it looks like maybe you're doing that (always creating a Manager) here?

Yes, but this is in `__call__`, which is only used when doing the `with ctx(event, state) as mgr` form.